std/range/package.d

```diff
@@ -1410,6 +1408,21 @@ if (isInputRange!(Unqual!R1) && isInputRange!(Unqual!R2) &&
     return Result(condition, r1, r2);
 }
 
+///
+unittest
+{
+    import std.algorithm.comparison : equal;
+
+    auto arr1 = [ 1, 2, 3, 4 ];
+    auto arr2 = [ 5, 6, 7, 8 ];
+
+    auto choice1 = choose(true, arr1, arr2);
```

Comment:
This is just `true? arr1 : arr2`. A better example chooses between two incompatible range types. Maybe apply `map` to `arr1` and `filter` to `arr2`.